xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/email/NotificationEmailUserPreferenceManager.java

```diff
@@ -36,8 +38,31 @@
     NotificationEmailDiffType getDiffType();
 
     /**
-     * @param userId id of a user
+     * @param userReference reference of a user
      * @return the diff type configured for the given user
+     * @since 14.10
      */
-    NotificationEmailDiffType getDiffType(String userId);
+    default NotificationEmailDiffType getDiffType(UserReference userReference)
```

Comment:
Missing `@Unstable` annotations in those new APIs.

Reply:
Right, I added `@Unstable`. Revapi is complaining about `getDiffType(String)` conversion to `getDiffType(UserReference)`, wondering what we should do about it.

Reply:
You should have kept `getDiffType(String)` but mark it as deprecated. And implement it by resolving the String to a UserReference.